stdlib/obj.mli

```diff
@@ -23,6 +23,23 @@ type t
 type raw_data = nativeint  (* @since 4.12 *)
 
 external repr : 'a -> t = "%identity"
+  (**
+     Internally, [repr] is just the identity function.
+     Using it has the effect of giving any value the abstract type [t].
+     One use is to keep values of different types in a single list
+     (perhaps to prevent them from being garbage collected).
```

Comment:
It doesn't sound like a good idea to suggest that one should use `Obj.repr` for a legitimate purpose ("keep values of different types in a single list"). Instead, I would document how to do this with existentials in the GADT section.